### PR TITLE
feature(wren-ai-service): refactor the query understanding pipe with async 

### PR DESCRIPTION
**Pull request by paopa**
This PR aims to refactor the pipeline with Hamilton to support async tasks for reducing the pending FastAPI request thread.

> [!NOTE]
> This is a breaking change because we have updated the OpenAI component to an asynchronous component. As a result, nearly all pipelines using the OpenAI generator will be affected. I suggest ensuring that the related test cases pass in the local environment and then merging the PR into the feature branch. so let's us skip the CI failed here.

## Compare the performance of synchronous and asynchronous LLM components.

- To try the synchronous LLM, check out the commit `ebce90a`.
- To try the asynchronous LLM, check out the commit `a2ce89e`.
- The Locust file is located in the `tools` folder. You can execute the following command to start Locust:
  ```bash
  poetry run locust -f ./tools/locust_pipe.py
  ```

### CODE REVIEW

File: wren-ai-service/pyproject.toml [hunk 1]
```diff
@@ -19,6 +19,7 @@ numpy = "==1.26.4"
 sqlparse = "==0.4.4"
 sqlglot = "==22.5.0"
 orjson = "==3.10.3"
+sf-hamilton = {extras = ["visualization"], version = "^1.63.0"}
```
Comment (cyyeh) [inline wren-ai-service/pyproject.toml:22]: please remember to use the fixed version in the future.

File: wren-ai-service/src/utils.py [hunk 2]
```diff
@@ -191,7 +191,7 @@ def timer(func):
 
     @functools.wraps(func)
     def wrapper_timer(*args, **kwargs):
-        if os.getenv("ENABLE_TIMER", False):
+        if os.getenv("ENABLE_TIMER", True):
```
[the remark this answers is not in the document]
Comment (paopa) [reply wren-ai-service/src/utils.py:194]: no, it should be false, just enable it for testing.

File: wren-ai-service/src/web/v1/routers.py [hunk 3]
```diff
@@ -123,3 +126,19 @@ async def get_ask_details_result(query_id: str) -> AskDetailsResultResponse:
     return container.ASK_DETAILS_SERVICE.get_ask_details_result(
         AskDetailsResultRequest(query_id=query_id)
     )
+
+
+dr = AsyncDriver({}, pipe)
+
+# dr = driver.Builder().with_modules(pipe).build()
```
[the remark this answers is not in the document]
Comment (paopa) [reply wren-ai-service/src/web/v1/routers.py:133]: AsyncDriver doesn't support the builder class. it belongs to the driver builder class.

File: wren-ai-service/pyproject.toml (hunk 1 above)
Comment (paopa) [reply wren-ai-service/pyproject.toml:22]: thanks for the reminder. this PR was published for testing purposes, so I think it still has lots of details that need to be modified. including this one. 😸

**Comment by cyyeh** [comment]
@paopa I would like to understand how would you implement the full pipeline rewrite using Hamilton AsyncDriver? Thanks!

**Comment by paopa** [comment]
> @paopa I would like to understand how would you implement the full pipeline rewrite using Hamilton AsyncDriver? Thanks!

sure. let us book a time to go thro the implementation details.

**Review by cyyeh — APPROVED** [review]
overall LGTM

**Pull request merged**